modules/schema/src/main/java/org/apache/ignite/internal/schema/NativeTypes.java

```diff
@@ -202,6 +219,12 @@ public static NativeType from(ColumnType type) {
                         ((ColumnType.VarLenColumnType)type).length() : Integer.MAX_VALUE
                 );
 
+            case NUMBER:
+                ColumnType.NumberColumnType numberType = (ColumnType.NumberColumnType)type;
+                if (numberType.precision() > 0)
+                    return new FixLenNumberNativeType(numberType.precision());
+                return new FixLenNumberNativeType(numberType.precision());
```

Comment:
Why do we need 'if' conditino here?

Reply:
I am sorry, request was not ready for review. Now it`s ready: I made many changes including fix for this 'if' condition.